torch/csrc/autograd/python_engine.cpp

```diff
@@ -81,7 +81,7 @@ void PythonEngine::thread_init(int device, const std::shared_ptr<ReadyQueue>& re
 
 #ifdef IS_PYTHON_3_9_PLUS
   // Do not call PyEval_RestoreThread, PyThreadState_[Clear|DeleteCurrent] if runtime is finalizing
-  if (_Py_IsFinalizing()) {
+  if (!Py_IsInitialized()) {
```

Comment:
If `Py_IsInitialized()` are used here, should `IS_PYTHON_3_9_PLUS` guard be removed?

Reply:
Will do in a follow up PR as discussed offline.

Reply:
https://github.com/pytorch/pytorch/issues/57507